plotly/plotly/plotly.py

```diff
@@ -1271,6 +1334,50 @@ def validate_credentials(credentials):
         raise exceptions.PlotlyLocalCredentialsError()
 
 
+def add_share_key_to_url(response, **payload):
+    """
+    Update plot's url to include the secret key
+
+    payload keyword agruments:
+    platform ('python') -- the platform that is associated with this url
+    version (string) -- the Plotly version that was used to create the plot
+    args (dict) -- contains the data from figure
+    un (string) -- username from credentials file
+    key (string) -- api_key from credentials file
+    origin('plot') --
+    kwargs (dict) --
+        'filename': (string) the name that is associated with this url
+        'fileopt': ('new' | 'overwrite' | 'extend' | 'append')
+                   file option of the url
+        'world_readable': (bool) make the plot public/ private
+        'sharing': ('public' | 'private' | 'secret')
+                    privacy option of the url
+        'layout': (dict) the layout of the figure
+
+    """
+    domain = (get_session_config().get('plotly_domain') or
+              get_config()['plotly_domain'])
+    plot_id = response['url'].replace(domain + "/~"
+                                      + payload['un'] + "/", "")
+    url = "https://api.plot.ly/v2/files/" + payload['un'] + ":" + plot_id
```

Comment:
to be more precise, get this from `get_config()` which talks to `session.py` to handle the settings.